dhis-2/dhis-support/dhis-support-artemis/src/main/java/org/hisp/dhis/artemis/audit/AuditableEntity.java

```diff
@@ -38,5 +38,19 @@
 @AllArgsConstructor
 public class AuditableEntity
 {
-    private Object entity;
+    /**
+     * Class of the AuditableEntity
+     * It will be used by {@link org.hisp.dhis.artemis.audit.legacy.AuditObjectFactory#collectAuditAttributes(Object)}
+     */
+    Class entityClass;
+
+    /**
+     * An object that is ready for serialized by Jackson.
+     * Means that this object should:
+     * 1. Only includes referenced properties that are owned by the current Audit Entity. Means that the property's schema has attribute "owner = true"
+     * 2. Do not include any lazy HibernateProxy or PersistentCollection that is not loaded.
+     * 3. All referenced properties that extend BaseIdentifiableObject should be mapped to only UID string
+     * This object could be a Map<String, Object>  with key is property name and value is the property value
+     */
+    Object serializableObject;
```

Comment:
I asked the same in @zubaira 's PR; Why are we using Object, and not Map (as commented) or Serializable?

Reply:
Map would make more sense as this  `serializableObject` is almost always Map.

Reply:
@stian-sandvold  @zubaira In some other places of the code some one could call AuditManager.send() and pass in any Java object.

This object will be serialized by `objectMapper.writeValueAsString(Object value)` which accepts any Java Object and serializes it to a String. 

I'm thinking of making another `createAuditEntry` funtion in another service where we have the object and session still open. So in that function we don't need to create HibernateProxy or open new session, just need to create an auditable object with correct format. With that , we can ensure to only use Map/Serializable for audit.